pkg/filters/args.go

```diff
@@ -42,14 +42,30 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 			}
 		}
 		if !found {
-			return true
+			return false // always filter if argument does not exist
 		}
-		// TODO: use type assertion instead of string conversion
+
 		if argName != "syscall" {
-			argVal = fmt.Sprint(argVal)
-		}
-		res := filter.Filter(argVal)
-		if !res {
+			switch argValue := argVal.(type) {
+			case string:
+				if !filter.Filter(argValue) {
+					return false
+				}
+				return true
+			case []string:
```

Comment:
We can implement this case much more efficiently with a FilterObject of its own. Internally just a map and a loop doing hash checks.

Reply:
Definitely. I was just exploring what was going on and checking what would brake what.

Reply:
If we do it as a generic object we could even cover other primitive array type (structs are more complex as can be seen in the SockAddr PR).
The problem is registering the correct filter at parse time since we have so many type names for arguments (I was working on resolving it but it's really hard to resolve to the decode type we use in `bufferdecoder`, since we have differing types for decoding and final representation)

Reply:
Keeping https://github.com/aquasecurity/tracee/issues/2486 opened and linking this comment there so we can discuss for the next release. For now I'll just skip those events we know for sure are broken. E2E tests are passing with the latest push, check if you agree pls. Then we can have at least a mitigation for this release.